### PR TITLE
[Enhancement] TwinTrim Website

### LINKED ISSUE #126: [Enhancement] Static site documentation
**Describe the enhancement**
Creating a full easy deployable & flexible markdown based documentation site with proper navbar items

**Why is this enhancement necessary?**
Having a proper documentation site for your project keep other in the loop of nay changes and guides for beginners for project setup

**Proposed solution**
Using an external package for creating building & deploying site using gh-pages

**Alternatives considered**
If applicable, mention any alternative solutions you've considered.

**Additional context**
Add any other context or screenshots related to the enhancement here.

### PR DESCRIPTION
**Pull request by yashksaini-coder**
## Description

This pull request includes significant updates to the documentation and workflow automation for the TwinTrim project. The most important changes include the addition of a GitHub Actions workflow to update contributors in the README, comprehensive documentation for features, usage, and contributing guidelines, and configuration for the MkDocs static site generator.

### Workflow Automation:
* Added a GitHub Actions workflow to automatically update the contributors list in the README file (`.github/workflows/contributors.yml`).

### Documentation:
* Added a detailed `Features.md` document describing key features such as duplicate detection, customizable filters, and multi-threaded processing (`docs/Features.md`).
* Created a `Usage.md` document with command-line interface instructions, options, examples, dependencies, and installation steps (`docs/Usage.md`).
* Added a `How-It-Works.md` document explaining the core components and key functions of the TwinTrim project (`docs/How-It-Works.md`).
* Updated `Contributing.md` with guidelines for contributing to the project (`docs/Contributing.md`).
* Configured MkDocs for generating a static site with navigation tabs, sections, and instant navigation (`mkdocs.yml`).

- Fixes #126 

## Type of change

A new feature add in the twintrim for its easy website of deploying & easy customization

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Tests update

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] I have maintained a clean commit history by using the necessary Git commands
- [X] I have checked that my code does not cause any merge conflicts

## Screenshots (if applicable)

Add screenshots to help explain the changes (if necessary).

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@Kota-Karthik can you review this PR

Here's the end result:- [site](https://yashksaini-coder.github.io/twinTrim/)

**Review by techy4shri — CHANGES_REQUESTED** [review]
Refer to repo docs for proper info and make the necessary changes.

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+To run TwinTrim, follow these steps:
+
+1. **Navigate to Project Directory**: Navigate to the project directory.
+2. **Run the Application**: Run the following command:
+
+   ```bash
+   python app.py
+   ```
+
+The application will start running on your local server. You can access it by navigating to `http://localhost:5000` in your web browser.
```
Comment (techy4shri) [inline docs/index.md:45]: you do not run it on server. This is a python package which is accessed via command line.

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    git push origin new-branch-name
+    ``` 
+
+5. **Create a Pull Request**: Create a pull request, from your fork to the main project.
+
+Your changes will be reviewed and merged into the main project.
+
+### Contact
+
+If you have any questions or need further assistance, please contact us at [Karthik Kota](https://www.linkedin.com/in/karthik-kota-44708924b/).
```
Comment (techy4shri) [inline docs/index.md:77]: please contact Karthik Kota
and then add link to the name.

**Review by techy4shri — CHANGES_REQUESTED** [review]
twinTrim is a python package, so you do not access it via localhost. Also, edit the contributing section and write what is written in the paragraph of Contribution guidelines.

**Comment by Kota-Karthik** [comment]
@yashksaini-coder LGTM!

**Comment by techy4shri** [comment]
Reverting this due to unresolved issues and errors.

**Comment by yashksaini-coder** [comment]
okh I will fix it, @techy4shri @Kota-Karthik

**Comment by techy4shri** [comment]
This PR is being rejected due to its lack of necessity, specifically as there is no requirement for a site generator in this project. Such contributions fall outside the scope of our current objectives. We will not be considering any other similar approaches.

**Pull request merged**